src/gluonnlp/model/train/language_model.py

```diff
@@ -103,8 +106,15 @@ def _get_decoder(self):
         output = nn.HybridSequential()
         with output.name_scope():
             if self._tie_weights:
-                output.add(nn.Dense(self._vocab_size, flatten=False,
-                                    params=self.embedding[0].params))
+                if self._shared_params is not None:
+                    shared_params = self.embedding[0].params
+                    shared_params = ParameterDict(shared_params.prefix)
+                    shared_params.update(self._shared_params)
```

Comment:
What is the difference between `shared_params` and `self.embedding[0].params` after these 3 lines?

Reply:
`shared_params` now shares the same prefix with `self.embedding[0].params`. `self._shared_params` has different `prefix` with `self.embedding[0].params`